src/core/api/registry.go

```diff
@@ -0,0 +1,210 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/replication/ng/model"
+	"github.com/goharbor/harbor/src/replication/ng/registry"
+)
+
+// RegistryAPI handles requests to /api/registries/{}. It manages registries integrated to Harbor.
+type RegistryAPI struct {
+	BaseController
+	manager registry.Manager
+}
+
+// Prepare validates the user
+func (t *RegistryAPI) Prepare() {
+	t.BaseController.Prepare()
+	if !t.SecurityCtx.IsAuthenticated() {
+		t.HandleUnauthorized()
+		return
+	}
+
+	if !t.SecurityCtx.IsSysAdmin() {
+		t.HandleForbidden(t.SecurityCtx.GetUsername())
+		return
+	}
+
+	t.manager = registry.NewDefaultManager()
+	if t.manager == nil {
+		log.Error("failed to create registry manager")
+		t.CustomAbort(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
+	}
+}
+
+// Get gets a registry by id.
+func (t *RegistryAPI) Get() {
+	id := t.GetIDFromURL()
+
+	registry, err := t.manager.Get(id)
+	if err != nil {
+		log.Errorf("failed to get registry %d: %v", id, err)
+		t.CustomAbort(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
+	}
+
+	if registry == nil {
```

Comment:
The registry will not be nil as the manager will return an error if it cannot get an valid registry from database? Maybe we need the manager returns nil when it cannot get the registry to handle the 404 error in API?

Reply:
I will return specific error to indicate not found.